Source/PeleLM.cpp

```diff
@@ -5819,36 +5776,21 @@ PeleLM::advance_chemistry (MultiFab&       mf_old,
 #endif
         BL_PROFILE_VAR_STOP(ReactInLoop);
 
-
-        /* Unpacking of data */
-        BL_PROFILE_VAR_START(ARRAY_FLATTEN);
-        amrex::ParallelFor(bx,
-        [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept
+        // Convert CGS -> MKS
+        ParallelFor(bx, [rhoY,rhoH, frc_rhoY, frc_rhoH]
+        AMREX_GPU_DEVICE (int i, int j, int k) noexcept 
         {
-           int icell = (k-lo.z)*len.x*len.y + (j-lo.y)*len.x + (i-lo.x);
-           gpu_unflatten(icell, i, j, k, rhoY, fcl,
-                         tmp_vect, tmp_vect_energy, tmp_fctCn);  
+           for (int n = 0; n < NUM_SPECIES; n++) {
+              rhoY(i,j,k,n) *= 1.0e3;
+              frc_rhoY(i,j,k,n) *= 1.0e3;
+           }
+           rhoH(i,j,k) *= 0.1;
+           frc_rhoH(i,j,k) *= 0.1;
         });
-        BL_PROFILE_VAR_STOP(ARRAY_FLATTEN);
 
-        /* Deallocate */
-        BL_PROFILE_VAR_START(Allocs);
-#ifdef USE_CUDA_SUNDIALS_PP 
+#ifdef USE_CUDA_SUNDIALS_PP
```

Comment:
Why is this sync needed?

Reply:
Maybe it isn't. Still unsure about what AMReX does and doesn't. We can either understand it or remove it and try it